internal/docstore/query.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docstore
+
+import (
+	"context"
+	"io"
+
+	"gocloud.dev/internal/docstore/driver"
+)
+
+// Query represents a query over a collection.
+type Query struct {
+	coll *Collection
+	dq   *driver.Query
+}
+
+// Query creates a new Query over the collection.
+func (c *Collection) Query() *Query {
+	return &Query{coll: c, dq: &driver.Query{}}
+}
+
+// Where expresses a condition on the query.
+// Valid ops are: "=", ">", "<", ">=", "<=".
+func (q *Query) Where(field, op string, value interface{}) *Query {
```

Comment:
s/field/fieldpath/

Call parseFieldPath.


The Query type should have an `err` field that remembers the error; it will be returned immediately by the  `Delete` and `Update` methods, and the `Get` method will copy it into the iterator.